### PR TITLE
feat(ci): use merge queues

### LINKED ISSUE #11253: ci: only run heavy CI jobs if PR is labeled with "ready"
Currently every commit pushed to a PR triggers all tests, including the heavy ones like rust-tests, simtests, turbo hierarchy e2e tests. This cloggs the CI runners. We should only run the heavy tests after the PR is labeled with "ready".

### PR DESCRIPTION
**Pull request by muXxer**
# Description of change

## Motivation

Heavy CI (Rust tests + simtests, split-cluster compatibility checks) currently runs on every non-draft PR push. Contributors pushing several commits in a row saturate the self-hosted runner pool with runs on work that isn't ready for that level of verification, starving other PRs and `develop` (#11253).

Two earlier approaches gated heavy tests behind explicit opt-ins — a "ready" label (#11263) and a PR-description checkbox (#11540) — but both required custom dispatcher machinery (`pull_request_target` + `workflow_dispatch`) with many edge cases: results not attached to the PR, no way to enforce that heavy tests passed before merge, and logic that can only be tested after merging.

GitHub's **merge queue** solves the same problem with built-in machinery: heavy tests run **exactly once per merge, pre-merge**, on the PR merged onto the latest `develop`, and are enforceable as a required status check.

## How it works

- `hierarchy.yml` and `pr_lint.yml` get a `merge_group` trigger. When a PR is queued, GitHub merges it (plus any PRs ahead of it) onto `develop` in a temporary `gh-readonly-queue/*` branch and requests checks on that result.
- The heavy jobs — the `rust-tests` chain in `_rust.yml` (incl. the two paths-filter jobs that only feed it), `split-cluster`, and `split-cluster-sync-check` — now run only on `merge_group` and on pushes to long-lived branches, no longer on `pull_request` events. `dorny/paths-filter` (pinned v4.0.1) supports `merge_group` natively (diffs `base_sha..head_sha`), so crate change filtering keeps working in the queue.
- A new plain job **`heavy-tests-status`** aggregates the heavy jobs' results and is the single check to mark as required. It has to be a plain `always()` job rather than requiring the nested heavy checks directly: on `pull_request` events the heavy jobs are skipped reusable-workflow calls, and a skipped `uses:` job never expands its nested check runs — a required check that never reports would keep PRs from entering the queue. This job always reports: trivially green on PRs, real results on merge groups and pushes.
- `Validate PR title` (required check) skips itself on `merge_group` runs — there is no PR to validate there, and a skipped check run counts as passed.
- The draft/`[run-ci]` gate, previously copied into 8 job conditions, is centralized on the `diff` job; all other jobs are gated via `needs.diff.result` or `diff` outputs. Behavior is unchanged: drafts skip CI unless the body contains `[run-ci]`.

## Behavior

- PR push → lint/format/typos/license/clippy/execution-cut as before; heavy tests no longer run.
- PR queued for merge → full CI including heavy tests on the merged result; failure removes the PR from the queue without touching `develop`.
- Push to `develop` / `devnet` / `testnet` / `mainnet` / release branches → unchanged (heavy tests still run). For queue-merged commits this duplicates the queue run; kept for now to also cover direct pushes — dropping it is a possible follow-up.
- Nightly → unchanged (calls the test workflows directly, unaffected by the event gates).

## Rollout (repo admin, after this merges)

1. In the `develop` ruleset, enable **Require merge queue**. Set the status-check timeout generously (simtests are slow; 120+ min) and consider allowing several PRs per merge group to amortize heavy runs.
2. Add **`heavy-tests-status`** (GitHub Actions) to the required status checks. The existing required checks stay unchanged.

## Known trade-offs

- The light jobs run twice per merged PR (once on the PR, once in the queue) — inherent to merge queues, cheap compared to the heavy suite no longer running on every push.
- `[run-ci]` on a draft still triggers light CI only; there is no way to run heavy tests on a PR before queueing (`_split_cluster.yml` still has `workflow_dispatch` for manual runs).
- `develop_ci_slack_report.yml` filters on the `develop` branch and won't report merge-queue failures; PR authors get GitHub's dequeue notification instead.

## Links to any relevant issues

fixes #11253

Supersedes the approaches explored in #11263 and #11540.

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration tests): `dprint check` and `actionlint` pass on the changed workflows.
- [ ] Patch-specific tests (correctness, functionality coverage): the `merge_group` flow can only be exercised once the queue is enabled on `develop`; verify with a test PR right after enabling it.

### CODE REVIEW

File: .github/workflows/hierarchy.yml [hunk 1]
```diff
@@ -110,22 +122,52 @@ jobs:
...
+  # It must report on pull_request events too — a required check that never
+  # reports would keep PRs from entering the merge queue — which is why this
+  # is a plain job with `always()` rather than a required check on the heavy
+  # jobs themselves: those are skipped reusable-workflow calls on PRs, and a
+  # skipped `uses:` job never expands its nested check runs.
+  heavy-tests-status:
+    needs:
+      - rust
+      - split-cluster
+      - split-cluster-sync-check
```
Comment (Thoralf-M) [inline .github/workflows/hierarchy.yml:163]: Should there also be `- diffs`?

If the diff job fails on a merge-group run (paths-filter flake, checkout error), all three other needs are skipped, the case matches nothing, and the required check goes green. Since the other required checks would also be skipped (and skipped counts as passed), a PR could merge with essentially no CI having run on the merged result

Comment (muXxer) [reply .github/workflows/hierarchy.yml:163]: added

**Review by Thoralf-M — APPROVED** [review]

**Review by alexsporn — APPROVED** [review]
